### DISCUSSION TITLE
Lambda is recognized as a test case

### DISCUSSION THREAD

**Discussion by AlexPetul** [pytest-dev/pytest#13089] [General]
Hi there! Let's say i want to define a variable:

```python
test_var = lambda: None
```

Pytest recognizes this as a valid test case (instance of class `Function`). I understand why, but is it really the correct behavior? I cannot use any assertions here anyway, so why this has to be a valid test case?

**Comment by kwist-sgr** [comment]
You can prevent pytest from discovering such functions by setting a boolean `__test__` attribute to `False`

```python
test_var = lambda: None
test_var.__test__ = False
```

**Comment by dongfangtianyu** [comment]
```python

def check(a,b):
    assert a == b

test_var = lambda : check(1,2)
```

Although it cannot use `assert` itself, it can be used to call other code containing `assert`.

Since Python can treat lambda as a function, pytest does not need to treat it differently.

**Reply by AlexPetul** [reply]
True, thanks for the answer :)
